src/libstd/ffi/c_str.rs

```diff
@@ -646,6 +647,24 @@ impl From<CString> for Vec<u8> {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl<T: Into<Vec<u8>>> TryFrom<T> for CString {
```

Comment:
```
[00:02:22] error[E0119]: conflicting implementations of trait `core::convert::TryFrom<_>` for type `ffi::c_str::CString`:
[00:02:22]    --> /checkout/src/libstd/ffi/c_str.rs:651:1
[00:02:22]     |
[00:02:22] 651 | / impl<T: Into<Vec<u8>>> TryFrom<T> for CString {
[00:02:22] 652 | |     type Error = NulError;
[00:02:22] 653 | |
[00:02:22] 654 | |     fn try_from(t: T) -> Result<CString, NulError> {
[00:02:22] 655 | |         CString::new(t)
[00:02:22] 656 | |     }
[00:02:22] 657 | | }
[00:02:22]     | |_^
[00:02:22]     |
[00:02:22]     = note: conflicting implementation in crate `core`
```

It is conflicting with `impl<T, U> TryFrom<U> for T where T: From<U>`, because `CString: Into<Vec<u8>>`, i.e. this overlaps with the identity transformation `CString: TryFrom<CString>`. 

Try to change it to `impl TryFrom<Vec<u8>> for CString` and `impl<'a> TryFrom<&'a [u8]> for CString`.

Reply:
Perhaps this should just be `TryFrom<Vec<u8>>` instead. Only having the slice conversion results in a potentially unwanted allocation in the case of:

```rust
let vec = vec![/* ... */];

let c_str: CString = vec.try_into()?;
```